src/topology/algebra/continuous_monoid_hom.lean

```diff
@@ -231,16 +233,93 @@ end⟩⟩
 
 variables {A B C D E}
 
-instance [t2_space B] : t2_space (continuous_monoid_hom A B) :=
+@[to_additive] instance [t2_space B] : t2_space (continuous_monoid_hom A B) :=
 (is_embedding A B).t2_space
 
-instance : topological_group (continuous_monoid_hom A E) :=
+@[to_additive] instance : topological_group (continuous_monoid_hom A E) :=
 let hi := is_inducing A E, hc := hi.continuous in
 { continuous_mul := hi.continuous_iff.mpr (continuous_mul.comp (continuous.prod_map hc hc)),
   continuous_inv := hi.continuous_iff.mpr (continuous_inv.comp hc) }
 
+@[to_additive] lemma continuous_comp [locally_compact_space B] :
+  continuous (λ f : continuous_monoid_hom A B × continuous_monoid_hom B C, f.2.comp f.1) :=
+(is_inducing A C).continuous_iff.2 $ (continuous_map.continuous_comp'.comp
+    ((is_inducing A B).prod_mk (is_inducing B C)).continuous)
+
+@[to_additive] lemma continuous_comp_left (f : continuous_monoid_hom A B) :
+  continuous (λ g : continuous_monoid_hom B C, g.comp f) :=
+(is_inducing A C).continuous_iff.2 $ f.to_continuous_map.continuous_comp_left.comp
+  (is_inducing B C).continuous
+
+@[to_additive] lemma continuous_comp_right (f : continuous_monoid_hom B C) :
+  continuous (λ g : continuous_monoid_hom A B, f.comp g) :=
+(is_inducing A C).continuous_iff.2 $ f.to_continuous_map.continuous_comp.comp
+  (is_inducing A B).continuous
+
+variables (E)
+
+/-- `continuous_monoid_hom _ f` is a functor. -/
+@[to_additive "`continuous_add_monoid_hom _ f` is a functor."]
+def comp_left (f : continuous_monoid_hom A B) :
+  continuous_monoid_hom (continuous_monoid_hom B E) (continuous_monoid_hom A E) :=
+{ to_fun := λ g, g.comp f,
+  map_one' := rfl,
+  map_mul' := λ g h, rfl,
+  continuous_to_fun := f.continuous_comp_left }
+
+variables (A) {E}
+
+/-- `continuous_monoid_hom f _` is a functor. -/
+@[to_additive "`continuous_add_monoid_hom f _` is a functor."]
+def comp_right {B : Type*} [comm_group B] [topological_space B]
+  [topological_group B] (f : continuous_monoid_hom B E) :
+  continuous_monoid_hom (continuous_monoid_hom A B) (continuous_monoid_hom A E) :=
+{ to_fun := λ g, f.comp g,
+  map_one' := ext (λ a, map_one f),
+  map_mul' := λ g h, ext (λ a, map_mul f (g a) (h a)),
+  continuous_to_fun := f.continuous_comp_right }
+
 end continuous_monoid_hom
 
 /-- The Pontryagin dual of `A` is the group of continuous homomorphism `A → circle`. -/
 @[derive [topological_space, t2_space, comm_group, topological_group, inhabited]]
 def pontryagin_dual := continuous_monoid_hom A circle
+variables {A B C D E}
+
+noncomputable instance : continuous_monoid_hom_class (pontryagin_dual A) A circle :=
+continuous_monoid_hom.continuous_monoid_hom_class
+
+namespace continuous_monoid_hom
+
+/-- `pontryagin_dual` is a functor. -/
+protected noncomputable def dual (f : continuous_monoid_hom A B) :
+  continuous_monoid_hom (pontryagin_dual B) (pontryagin_dual A) :=
+f.comp_left circle
```

Comment:
I am not a big fan of this name (unless it's standard in the litterature), what do you think about `pontryagin_dual.map` (which means no dot notation) or `continuous_monoid_hom.transpose`? Again I don't know what is used in the literature, so if you prefer the current name it's fine.